app/api/decorators.py

```diff
@@ -91,7 +94,7 @@ def decorator(*args, **kwargs):
         if current_user.is_authenticated:
             return func(*args, **kwargs)
         try:
-            if data and data.get('api_key'):
+            if config.AUTH_METHOD == 'local' and data and data.get('api_key'):
```

Comment:
Please add `'local'` and other `AUTH_METHOD` constants to a new file in `app/constants` and import the string value from there